### DISCUSSION TITLE
DeDRE and Kortext

### DISCUSSION THREAD

**Discussion by paszczakus91** [apprenticeharper/DeDRM_tools#1950] [Q&A]
Hi,
I want to ask, it is any possibility to remove DRE from pdf downloaded via Kortext app?
ADE 2.0.1 authorize pdf but doesn't open it
ADE 3.0 and ADE 4.5 authorize and open PDF

> calibre 5.32 [64bit]  embedded-python: True is64bit: True
> Windows-10-10.0.19041 Windows ('64bit', 'WindowsPE')
> ('Windows', '10', '10.0.19041')
> Python 3.8.5
> Windows: ('10', '10.0.19041', '', 'Multiprocessor Free')
> Interface language: pl
> Successfully initialized third party plugins: DeDRM (7, 2, 1) && Obok DeDRM (7, 2, 1)
> calibre 5.32 [64bit]  embedded-python: True is64bit: True
> Windows-10-10.0.19041 Windows ('64bit', 'WindowsPE')
> ('Windows', '10', '10.0.19041')
> Python 3.8.5
> Windows: ('10', '10.0.19041', '', 'Multiprocessor Free')
> Interface language: pl
> Successfully initialized third party plugins: DeDRM (7, 2, 1) && Obok DeDRM (7, 2, 1)
> Turning on automatic hidpi scaling
> devicePixelRatio: 1.0
> logicalDpi: 96.0 x 96.0
> physicalDpi: 94.14671814671814 x 94.07407407407408
> Using calibre Qt style: True
> [0.00] Starting up...
> [0.02] Showing splash screen...
> [0.52] splash screen shown
> [0.52] Initializing db...
> [0.52] db initialized
> [0.52] Constructing main UI...
> DEBUG:    0.0 obok::utilities.py - loading translations
> DEBUG:    0.0 obok::dialogs.py - loading translations
> DEBUG:    0.0 obok::config.py - loading translations
> DEBUG:    0.0 obok::action_err.py - loading translations
> Key 'S' for shortcut Zapisz na dysku is already used by Scal wpisy książki, ignoring
> [1.16] main UI initialized...
> [1.16] Hiding splash screen
> Starting QuickView
> DeDRM v7.2.1: Trying to decrypt PDF_FILE.pdf
> DeDRM v7.2.1: PDF_FILE.pdf is a PDF ebook
> DeDRM v7.2.1: Looking for new default Adobe Digital Editions Keys after 0.1 seconds
> Found 1 keys
> DeDRM v7.2.1: Trying a new default key
> DeDRM v7.2.1: Exception when decrypting after 0.1 seconds
> Traceback (most recent call last):
>   File "calibre_plugins.dedrm.__init__", line 487, in PDFDecrypt
>   File "calibre_plugins.dedrm.ineptpdf", line 2178, in decryptBook
>   File "calibre_plugins.dedrm.ineptpdf", line 1984, in __init__
>   File "calibre_plugins.dedrm.ineptpdf", line 1469, in initialize
>   File "calibre_plugins.dedrm.ineptpdf", line 1599, in initialize_ebx
>   File "calibre_plugins.dedrm.ineptpdf", line 390, in decrypt
>   File "Crypto\Cipher\PKCS1_v1_5.py", line 165, in decrypt
> ValueError: Ciphertext with incorrect length.
> DeDRM v7.2.1: Failed to decrypt with new default key after 0.1 seconds
> DeDRM v7.2.1: Ultimately failed to decrypt after 0.1 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
> Running file type plugin DeDRM failed with traceback:
> Traceback (most recent call last):
>   File "calibre\customize\ui.py", line 175, in _run_filetype_plugins
>   File "calibre_plugins.dedrm.__init__", line 640, in run
>   File "calibre_plugins.dedrm.__init__", line 515, in PDFDecrypt
> calibre_plugins.dedrm.DeDRMError: DeDRM v7.2.1: Ultimately failed to decrypt after 0.1 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
> Traceback (most recent call last):
>   File "calibre\customize\ui.py", line 432, in get_file_type_metadata
>   File "calibre\customize\builtins.py", line 318, in get_metadata
>   File "calibre\ebooks\metadata\pdf.py", line 124, in get_metadata
> ValueError: Could not read info dict from PDF
> Syntax Error: Couldn't find the 'EBX_HANDLER' security handler
> 
> pdfinfo errored out with return code: 1
> Exception ignored in: <function NativeFunction.__del__ at 0x000001D36F50EA60>
> Traceback (most recent call last):
>   File "calibre_plugins.dedrm.adobekey", line 268, in __del__
> TypeError: 'NoneType' object is not callable
> Exception ignored in: <function NativeFunction.__del__ at 0x000001D36F50EA60>
> Traceback (most recent call last):
>   File "calibre_plugins.dedrm.adobekey", line 268, in __del__
> TypeError: 'NoneType' object is not callable
> [32.33] splash screen hidden
> [32.33] Started up in 32.33 seconds with 1 books
> stdout+stderr from file dialog helper: [b'', b'']
> piped data from file dialog helper: [b'`\x8fi\xf5\\\xf1\xaf\xb72v\xf1O\xf3\xecMN\xd2oI*\xc3O\t\xde\xef@~\x7f\xf3\x0ec\x89', b'C:\\Users\\admin\\AppData\\Local\\Packages\\Kortext.Kortext_htp4jyd3mpprc\\LocalState\\Library\\PDF_FILE.pdf']
> Added PDF_FILE to db in: 0.1
> Added 1 books in 0.8 seconds

**Comment by Leseratte10** [comment]
This looks like a PDF with the new Adobe DRM. 
Did you initially download it with ADE 2 or ADE3+?

There's some books that require ADE 3, they can't be opened in ADE 2 (or have their DRM removed) at all. 
And there's also books that can be opened in ADE 2, but if you open them in ADE 3 once they'll be permanently locked to ADE 3.

**Reply by paszczakus91** [reply]
Initially, I downloaded it with ADE 4.5, but I will try on another system with ADE 2

**Reply by paszczakus91** [reply]
On fresh Windows I tried to open file via ADE 2 with error (but after successful authorization)
In Calibre I received the same error https://github.com/apprenticeharper/DeDRM_tools/discussions/1950#discussioncomment-1728283

**Comment by ZolaLa9** [comment]
@paszczakus91 - Have you tried [v10.0.2](https://github.com/noDRM/DeDRM_tools/releases) maintained by @noDRM? (Apprentice Harper is [is no longer maintaining these tools](https://github.com/apprenticeharper/DeDRM_tools#readme).)

I suspect v10.0.2 will not work either, however noDRM has made a [variety of changes](https://github.com/noDRM/DeDRM_tools/blob/master/CHANGELOG.md) to the ADE process, including adding "more verbose error message when trying to remove DRM from a book with the new, not-yet-cracked version of the Adobe ADEPT DRM".

**Reply by paszczakus91** [reply]
No, i didn't try it, but I will :)

**Reply by paszczakus91** [reply]
I received:

```
DeDRM v10.0.2: Trying a new default key
This seems to be an Adobe ADEPT PDF with Adobe's new DRM
This DRM cannot be removed yet. 
Try getting your distributor to give you a new ACSM file, then open that in an old version of ADE (2.0).
If your book distributor is not enforcing the new DRM yet, this will give you a copy with the old DRM.
DeDRM v10.0.2: Exception when decrypting after 0.3 seconds
```

**Reply by ZolaLa9** [reply]
Thanks for reporting back. 10.0.x is so new I was curious what the "more verbose message" would produce :)
